cornell-dti/office-hours
# New upload roles design

### Summary 



This PR implements a new design to upload roles. On the Manage Roles page of the Dashboard, the new "Import Professors/TAs" button opens a modal, where the user can choose to upload roles manually or upload a file. The CSV upload option allows professors to select or drag/drop a .xlsx or .csv file with roles to upload. The selected file should only be shown if it is in the correct format. If not, an error is displayed and the selected file is not shown. To upload manually, emails and roles can be entered into a table. Roles can be uploaded once a valid email is entered for every row and at least one email and role is entered. Once roles are uploaded correctly, the modal closes, and a confirmation message is shown. 






### Test Plan 


CSV Upload
- Check that files can be uploaded by using both select and drag/drop options.
- Check that an error is displayed and the uploaded file is not shown if the file is not in .xlsx or .csv format.
- Check that an error is displayed and the uploaded file is not shown if the file contains roles other than the allowed roles (TA, Professor).
- If the uploaded file has the correct format, the file name should be shown in the box.
- After a valid file is uploaded and the confirm button is clicked, the modal should close and a message should show pending roles.
- The CSV upload modal should reset once the modal is closed. Check that when the modal is exited and reopened, no file or error messages are shown. 

Manual Upload
- Check that the delete icons delete the corresponding row in the table.
- Check that rows can be added using the add icon. 
- Check that if several rows are added, all rows can be viewed by scrolling vertically. 
- Check that an error is displayed and the confirm button does not work if there is an invalid email or empty email cell. 
- Check that an error is displayed and the confirm button does not work if there are no rows in the table.
- After roles are entered in the correct format and the confirm button is clicked, the modal should close and a message should show pending roles.
- The manual upload modal should reset once the modal is closed. Check that when the modal is exited and reopened, the default three rows are shown. 
### Notes 



### Breaking Changes 

None
